zerocopy-derive/tests/struct_known_layout.rs

```diff
@@ -46,16 +48,37 @@ util_assert_impl_all!(TypeParams<'static, (), imp::IntoIter<()>>: imp::KnownLayo
 util_assert_impl_all!(TypeParams<'static, util::AU16, imp::IntoIter<()>>: imp::KnownLayout);
 
 // Deriving `KnownLayout` should work if the struct has bounded parameters.
+//
+// N.B. We limit this test to rustc >= 1.62, since earlier versions of rustc ICE
+// when `KnownLayout` is derived on a `repr(C)` struct whose trailing field
+// contains non-static lifetimes.
+#[rustversion::since(1.62)]
+const _: () = {
+    #[derive(imp::KnownLayout)]
+    #[repr(C)]
+    struct WithParams<'a: 'b, 'b: 'a, T: 'a + 'b + imp::KnownLayout, const N: usize>(
+        [T; N],
+        imp::PhantomData<&'a &'b ()>,
+    )
+    where
+        'a: 'b,
+        'b: 'a,
+        T: 'a + 'b + imp::KnownLayout;
+
+    util_assert_impl_all!(WithParams<'static, 'static, u8, 42>: imp::KnownLayout);
+};
```

Comment:
Also add a test which is similar but uses static lifetimes and is compiled on all toolchains?

Reply:
[Done](https://github.com/google/zerocopy/compare/c7cfc8df03d86f6f93d73c8bc37729ad12eebd96..2cfcc5a8c3baf2a66dbb18fe633ba234a411a155#diff-85b159abf94564dfae9563267afd74a8b55832be2174e5cfef9522c1a8b6955cR71-R87)